### PR TITLE
Precommit

### PR DESCRIPTION
**Pull request by fgoiriz**
## Summary

added pre-commit action and hook that will be run on every Pull Request to viadot

## Importance

To add pre-commit checks on every PR in viadot

## Checklist

This PR:

- [] adds new tests (if appropriate)
- [] updates docstrings for any new functions or function arguments (if appropriate)
- [x] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

**Comment by Trymzet** [comment]
@fgoiriz pls also add documentation in CONTRIBUTING.md

File: .github/workflows/static_analysis.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
```
Comment (Trymzet) [inline .github/workflows/static_analysis.yml:18]: We're on 3.10, can we use that one?

File: .github/workflows/static_analysis.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Static analysis
+
+on: [pull_request]
```
Comment (Trymzet) [inline .github/workflows/static_analysis.yml:3]: Should only be branch 2.0 for now

File: .github/workflows/build-2.0.yml [hunk 3]
```diff
@@ -4,10 +4,27 @@ on:
...
+        env:
+          VIADOT_CONFIG_PATH: .config/credentials.json.template
+        run: |
+          pip install pytest
+          sudo apt install libsqliteodbc
+          pytest tests/unit
+
   publish-docker-image:
     runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/dev' && github.event_name == 'push'
```
Comment (Trymzet) [inline .github/workflows/build-2.0.yml:27]: This is wrong, we're not dealing with the `dev` branch here

**Review by Trymzet — CHANGES_REQUESTED** [review]
Added my comments

File: .github/workflows/build-2.0.yml (hunk 3 above)
Comment (fgoiriz) [reply .github/workflows/build-2.0.yml:27]: Changed it to 2.0

File: .github/workflows/static_analysis.yml (hunk 2 above)
Comment (fgoiriz) [reply .github/workflows/static_analysis.yml:3]: ok

**Comment by fgoiriz** [comment]
Pushed changes, should we change also the destination of this PR from viadot:main to viadot:2.0 ?

File: .github/workflows/build-2.0.yml (hunk 3 above)
Comment (Trymzet) [reply .github/workflows/build-2.0.yml]: This action already only applies to 2.0 so this line is not needed

**Review by Trymzet — CHANGES_REQUESTED** [review]
Requested some changes in the existing conversations; for the future, pls don't resolve them yourself

**Pull request closed without merging**